jsonapi_requests/request_factory.py

```diff
@@ -67,9 +67,11 @@ def configured_options(self):
             options['timeout'] = self.config.TIMEOUT
         return options
 
-    def _parse_response(self, response):
+    def _parse_response(self, response, method):
         if response.status_code >= 500:
             raise ApiInternalServerError(response.status_code, response.content)
+        elif method == 'DELETE' and response.status_code == 204:
```

Comment:
I am not sure if it is  good idea to check method on this level. 204 can also be returned on  other methods (I am sure POST can return 204)